### ISSUE TITLE
Getting black images as ouput on current data

### ISSUE THREAD

**Issue by purijs** [idealo/image-super-resolution#153]
I'm using the following for training on my custom `png`

```
from ISR.models import RRDN
from ISR.models import Discriminator
from ISR.models import Cut_VGG19

lr_train_patch_size = 32
layers_to_extract = [5, 9]
scale = 4
hr_train_patch_size = lr_train_patch_size * scale

rrdn  = RRDN(arch_params={'C':4, 'D':3, 'G':64, 'G0':64, 'T':10, 'x':scale}, patch_size=lr_train_patch_size)
f_ext = Cut_VGG19(patch_size=hr_train_patch_size, layers_to_extract=layers_to_extract)
discr = Discriminator(patch_size=hr_train_patch_size, kernel_size=3)

from ISR.train import Trainer
loss_weights = {
  'generator': 0.0,
  'feature_extractor': 0.0833,
  'discriminator': 0.01
}
losses = {
  'generator': 'mae',
  'feature_extractor': 'mse',
  'discriminator': 'binary_crossentropy'
}

log_dirs = {'logs': './logs', 'weights': './weights'}
learning_rate = {'initial_value': 0.0004, 'decay_factor': 0.5, 'decay_frequency': 30}
flatness = {'min': 0.0, 'max': 0.15, 'increase': 0.01, 'increase_frequency': 5}


trainer = Trainer(
    generator=rrdn,
    discriminator=discr,
    feature_extractor=f_ext,
    lr_train_dir='data/lr/t/',
    hr_train_dir='data/hr/t/',
    lr_valid_dir='data/lr/v/',
    hr_valid_dir='data/hr/v/',
    loss_weights=loss_weights,
    learning_rate=learning_rate,
    flatness=flatness,
    dataname='image_dataset',
    log_dirs=log_dirs,
    n_validation=10,
)

model=trainer.train(
    epochs=10,
    steps_per_epoch=100,
    batch_size=1,
    monitored_metrics={'val_PSNR_Y': 'max'}
)
```
After training, this is how I'm predicting, I'm not really sure how to predict here because 'trainer' as no predict attribute
![image](https://user-images.githubusercontent.com/22521404/96685858-54e3b580-139b-11eb-8250-194bae5b5570.png)

I tried exporting weights also using `rrdn.model.save_weights('./weights/sp.hdf5')` but get the same output

Is this some problem with my training? My HR images are `128x128` and LR is `32x32`

**Comment by andre-fu** [comment]
Also getting black images as outputs on inputs. Any solution this this? @purijs

**Comment by andre-fu** [comment]
I think the issue is with the scaling of images. De-normalize and scale to the 0-255 range (instead of 0-1) and run the imsave as normal.

**Comment by sephethus** [comment]
> I think the issue is with the scaling of images. De-normalize and scale to the 0-255 range (instead of 0-1) and run the imsave as normal.

No idea what this means.

**Comment by andre-fu** [comment]
@sephethus matplotlib has uses image scaling on the 0-255 range while the image that gets generated is using 0-1 range. Multiply the image by 255 to scale the image to the correct input for matplotlib and it will work.

**Comment by sephethus** [comment]
Guys I'm doing the same thing here with custom datasets and I can't even figure out where the results are being written to after training finishes.

**Comment by waleedrazakhan92** [comment]
> Thanks @andre-fu
> 
> Guys I'm doing the same thing here with custom datasets and I see that several different hdf5 files were written to the weights dir under the time/date stamp for today and I'm trying to figure out which one to use. Any ideas what the next steps are here? I can't even get to the problem you guys are having yet.

Use the one with the latest epoch number. I trained the rrdnn model so the generator weights were saved with the rrdnn* and the disriminator weights were stored with the name srgan*. For prediction use only the generator weights. You can check the name for the weights by printing out trainer.helper.(name of the model you want to see)
